lib/gds_api/email_alert_api.rb

```diff
@@ -267,6 +267,17 @@ def bulk_unsubscribe(slug:, govuk_request_id: nil, body: nil, sender_message_id:
     )
   end
 
+  # Update subscriber list details, such as title
+  # @param [Hash]         params        A hash of detail paramaters that can be updated. For example title.
+  #                                     For allowed parameters see:
+  #                                     https://github.com/alphagov/email-alert-api/blob/main/docs/api.md#patch-subscriber-listsxxx
```

Comment:
I weakly feel that you should take the actual parameters here.  It means we have to update gds-api-adapters when those change; but if it's ok in this case to just take an arbitrary hash and say "consult the API docs!" why don't all of the gds-api-adapters methods do that?

Reply:
Yeah, I see what you are saying, I came down on the other side of this after a bit of thinking. Partly with one eye to the upcoming description addition.

I figured that there is very little parameter validation going on in the gem. If the API can return a status code telling me what's a valid / useful slug, then it's the APIs job to say what params it's permitting to update.

GDS API adapters is just to act as helpers to facilitate that right?
So as long as there are useful test helpers for what happens if the API rejects you with 422.

Only thing explicitly stating the params here would do is mean we have two places to update / more minor version bumps if we added other params? Can't think of a scenario where that'd help?

Reply:
Perhaps especially as if you were sending params that weren't permitted to the api, they'll just be dropped, and it'll error if empty?

I basically don't care what else is in there as long as one valid param exists?